platform/posix/transport/src/openssl_posix.c

```diff
@@ -535,7 +541,7 @@ OpensslStatus_t Openssl_Connect( NetworkContext_t * pNetworkContext,
     }
 
     /* Clean up on error. */
-    if( ( returnStatus != OPENSSL_SUCCESS ) && ( sslStatus == 0 ) )
+    if( ( returnStatus != OPENSSL_SUCCESS ) && ( sslObjectCreated == 1u ) )
```

Comment:
We do `SSL_CTX_new` at line 453. Do we not need to free it using `SSL_CTX_free`? We should use valgrind to see if we are leaking memory.

Reply:
+1 about valgrind

Reply:
The allocated memory `pSslContext` with `SSL_CTX_new` is freed on line 534. As that is a local variable to the function, it does not need a flag to determine if it needs to be freed, a simple NULL check is sufficient for it.